### PR TITLE
Prefer make over druidic knowledge

### PR DESCRIPTION
**Pull request by ignasi35**
Getting up and running with the project requires being familiar with the Go ecosystem or the GitHub Actions YAML.

By moving all the build steps to a makefile it becomes a bit more universal and easy for noobs (like me) to get their local env up and running.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,8 @@
+# Contributing
```
Comment (ignasi35) [inline CONTRIBUTING.md:1]: Related to #4

**Review by sdecandelario — APPROVED** [review]

File: Makefile [hunk 2]
```diff
@@ -1,2 +1,19 @@
...
+	go get -v -t -d ./...
+
+fast-build:
+	go build -v .
+
+build: clean get-dependencies fast-build
+	echo "'candebot' built and ready to serve and protect."
+
+test: fast-build
+	go test -v ./...
```
Comment (gonzaloserrano) [inline Makefile:19]: nit: missing EOF (configurable by your IDE)

**Review by gonzaloserrano — APPROVED** [review]
LGTM

File: Makefile (hunk 2 above)
Comment (ignasi35) [reply Makefile]: Done, and config'd my editor aswell. Thanks for the callout!

**Comment by ignasi35** [comment]
May collision with #61

File: Makefile [hunk 3]
```diff
@@ -1,2 +1,19 @@
...
+
+get-dependencies:
+	echo "Getting dependencies..."
+	go get -v -t -d ./...
+
+fast-build:
+	go build -v .
+
+build: clean get-dependencies fast-build
+	echo "'candebot' built and ready to serve and protect."
```
Comment (gonzaloserrano) [inline Makefile:16]: The get-deps step is not needed, as go automatically deals with that when building or installing. See https://github.com/bcneng/candebot/pull/61.

So I think we can simplify to:

```suggestion
build: clean
	go build -v .
	@echo candebot built and ready to serve and protect.
```

p.s notice I prepended the echo with `@` so it's not printed twice, see https://www.gnu.org/software/make/manual/html_node/Echoing.html

also there is no var interpolation so no double quoting needed I guess

File: Makefile [hunk 4]
```diff
@@ -1,2 +1,19 @@
...
+
+test: fast-build
```
Comment (gonzaloserrano) [inline Makefile:18]: no need to depend on build since test creates a binary and handles deps too
```suggestion
test:
```

Comment (ignasi35) [reply Makefile:12]: TIL

**Comment by ignasi35** [comment]
> May collision with #61

Not anymore (see following commits on this PR).

**Review by ignasi35 — COMMENTED** [review]
LGTM

File: Makefile [hunk 5]
```diff
@@ -1,2 +1,13 @@
 lint:
 	golangci-lint run
+
+clean:
+    # '-f' to ignore when 'candebot' is not found
```
Comment (gonzaloserrano) [inline Makefile:5]: this line must be indented by a tab, not spaces

Comment (ignasi35) [reply Makefile:5]: Good call

**Pull request merged**